### PR TITLE
# 167879181 Connects several endpoints to the database

### PR DESCRIPTION
**Pull request by nziokaivy**
#### What does this PR do?
Connects several endpoints to the database

#### Description of Task to be completed?
- Connect cancel trip endpoint to the database
- Connect get all trips endpoint to the database
- Connect get specific trip endpoint to the database
-  Connect create new booking endpoint to the database
- Connect get all bookings endpoint to the database
-  Connect get bookings by made by user bookings endpoint to the database
-  Connect get delete endpoint to the database

#### How should this be manually tested?
Clone this repository and run `npm install && run test`

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
#167879181

#### Screenshots (if appropriate)
Screenshot from 2019-08-15 05-48-17
Screenshot from 2019-08-15 05-49-08
Screenshot from 2019-08-15 06-09-26
Screenshot from 2019-08-15 05-50-35
Screenshot from 2019-08-15 05-51-13
Screenshot from 2019-08-15 06-11-11
Screenshot from 2019-08-15 06-11-48

### CODE REVIEW

File: server/api/controllers/tripController.js [hunk 1]
```diff
@@ -22,47 +22,55 @@ class TripController {
...
 
-	static cancelTrip(req, res) {
+	static async cancelTrip(req, res) {
 		const id = parseInt(req.params.id);
-		const cancelTrip = Trip.cancelTrip(id);
-		if (!cancelTrip) {
-			return res.status(404).json({
-				status: 'error',
-				error: 'Not found',
+		// eslint-disable-next-line prefer-destructuring
```
Comment (oesukam) [inline server/api/controllers/tripController.js]: Destruct and remove this inline eslint rule

File: server/api/db/Db.js [hunk 2]
```diff
@@ -1,8 +1,10 @@
 import { Pool } from 'pg';
 // eslint-disable-next-line no-unused-vars
```
Comment (oesukam) [inline server/api/db/Db.js]: Not necessary

File: server/api/models/booking.js [hunk 3]
```diff
@@ -1,105 +1,89 @@
...
-			email: 'janedoe@gmail.com',
-			seat_number: 4,
-		},
-		];
-	}
+	// eslint-disable-next-line no-useless-constructor
+	constructor() {}
 
-	createNewBooking({
+	// eslint-disable-next-line class-methods-use-this
```
Comment (oesukam) [inline server/api/models/booking.js]: Add it in the eslint configuration's rule instead

File: server/api/models/booking.js [hunk 4]
```diff
@@ -1,105 +1,89 @@
...
-			email,
-
-		};
-		this.bookings.push(newBooking);
-		return newBooking;
 	}
 
-	getAllBookings() {
-		return this.bookings;
+	// eslint-disable-next-line class-methods-use-this
```
Comment (oesukam) [inline server/api/models/booking.js]: Goes in the eslint config file

File: server/api/models/trip.js [hunk 5]
```diff
@@ -1,47 +1,104 @@
+/* eslint-disable no-shadow */
+/* eslint-disable no-else-return */
+/* eslint-disable no-restricted-globals */
+/* eslint-disable no-unused-vars */
```
Comment (oesukam) [inline server/api/models/trip.js]: Not necessary

File: server/api/models/trip.js [hunk 6]
```diff
@@ -1,47 +1,104 @@
+/* eslint-disable no-shadow */
+/* eslint-disable no-else-return */
```
Comment (oesukam) [inline server/api/models/trip.js]: To be put in the eslint config file

**Review by oesukam — CHANGES_REQUESTED** [review]

File: server/api/models/trip.js (hunk 6 above)
Comment (nziokaivy) [reply server/api/models/trip.js]: Done

File: server/api/models/trip.js (hunk 5 above)
Comment (nziokaivy) [reply server/api/models/trip.js]: I have now removed them

File: server/api/models/booking.js (hunk 4 above)
Comment (nziokaivy) [reply server/api/models/booking.js]: I have added to the eslint config file

File: server/api/models/booking.js (hunk 3 above)
Comment (nziokaivy) [reply server/api/models/booking.js]: Added to the eslint config file

File: server/api/controllers/tripController.js (hunk 1 above)
Comment (nziokaivy) [reply server/api/controllers/tripController.js]: Removed

File: server/api/db/Db.js (hunk 2 above)
Comment (nziokaivy) [reply server/api/db/Db.js]: Removed

**Review by oesukam — APPROVED** [review]

**Pull request merged**